File: 1.0/src/c/cz/mg/vulkan/PFN_vkCmdWriteBufferMarkerAMD.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdWriteBufferMarkerAMD_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong pipelineStage, jlong dstBuffer, jlong dstOffset, jlong marker){
    (void)env;
    (void)jc;
    PFN_vkCmdWriteBufferMarkerAMD f = (PFN_vkCmdWriteBufferMarkerAMD)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkPipelineStageFlagBits*)jniLongToPointer(pipelineStage)),
        *((VkBuffer*)jniLongToPointer(dstBuffer)),
        *((VkDeviceSize*)jniLongToPointer(dstOffset)),
        *((uint32_t*)jniLongToPointer(marker))
    );
}


void Java_cz_mg_vulkan_PFNvkCmdWriteBufferMarkerAMD_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jint pipelineStage, jlong dstBuffer, jlong dstOffset, jint marker){
    (void)env;
    (void)jc;
    PFN_vkCmdWriteBufferMarkerAMD f = (PFN_vkCmdWriteBufferMarkerAMD)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        (VkPipelineStageFlagBits)pipelineStage,
        *((VkBuffer*)jniLongToPointer(dstBuffer)),
        (VkDeviceSize)dstOffset,
        (uint32_t)marker
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkCmdWriteBufferMarkerAMD.java
package cz.mg.vulkan;

public class PFNvkCmdWriteBufferMarkerAMD extends VkFunctionPointer {
    public PFNvkCmdWriteBufferMarkerAMD() {
    }

    protected PFNvkCmdWriteBufferMarkerAMD(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdWriteBufferMarkerAMD(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdWriteBufferMarkerAMD(long value) {
        setValue(value);
    }

    public PFNvkCmdWriteBufferMarkerAMD(VkInstance instance) {
        super(instance, new VkString("vkCmdWriteBufferMarkerAMD"));
    }

    public void call(VkCommandBuffer commandBuffer, VkPipelineStageFlagBits pipelineStage, VkBuffer dstBuffer, VkDeviceSize dstOffset, VkUInt32 marker){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), pipelineStage != null ? pipelineStage.getVkAddress() : VkPointer.getNullAddressNative(), dstBuffer != null ? dstBuffer.getVkAddress() : VkPointer.getNullAddressNative(), dstOffset != null ? dstOffset.getVkAddress() : VkPointer.getNullAddressNative(), marker != null ? marker.getVkAddress() : VkPointer.getNullAddressNative());
    }

    protected static native void callNative(long vkaddress, long commandBuffer, long pipelineStage, long dstBuffer, long dstOffset, long marker);


    public void call(VkCommandBuffer commandBuffer, int pipelineStage, VkBuffer dstBuffer, long dstOffset, int marker){
        callSimplifiedNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), pipelineStage, dstBuffer != null ? dstBuffer.getVkAddress() : VkPointer.getNullAddressNative(), dstOffset, marker);
    }

    protected static native void callSimplifiedNative(long vkaddress, long commandBuffer, int pipelineStage, long dstBuffer, long dstOffset, int marker);

}
